### PR TITLE
lib/promscrape: allows to use k8s pod name as clusterMemberNum

### PR DESCRIPTION
**Pull request by f41gh7**
it must improve user expirience and simplify clustering scrapers.
it must allow to use vmagent cluster with distroless images
https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2359

### CODE REVIEW

**Review by hagen1778 — APPROVED** [review]
LGTM, but build fails

File: lib/promscrape/config.go [hunk 1]
```diff
@@ -49,12 +49,32 @@ var (
 	clusterMembersCount = flag.Int("promscrape.cluster.membersCount", 0, "The number of members in a cluster of scrapers. "+
 		"Each member must have an unique -promscrape.cluster.memberNum in the range 0 ... promscrape.cluster.membersCount-1 . "+
 		"Each member then scrapes roughly 1/N of all the targets. By default cluster scraping is disabled, i.e. a single scraper scrapes all the targets")
-	clusterMemberNum = flag.Int("promscrape.cluster.memberNum", 0, "The number of number in the cluster of scrapers. "+
-		"It must be an unique value in the range 0 ... promscrape.cluster.membersCount-1 across scrapers in the cluster")
+	clusterMemberNum = flag.String("promscrape.cluster.memberNum", "0", "The number of number in the cluster of scrapers. "+
+		"It must be an unique value in the range 0 ... promscrape.cluster.membersCount-1 across scrapers in the cluster. "+
+		"Can be specified as pod name of kubernetes statefulset - pod-name-1, where Num is numeric part of pod after -")
```
Comment (valyala) [inline lib/promscrape/config.go:54]:
```suggestion
		"Can be specified as pod name of Kubernetes StatefulSet - pod-name-Num, where Num is a numeric part of pod name")
```

File: lib/promscrape/config.go [hunk 2]
```diff
@@ -49,12 +49,32 @@ var (
...
 
+var clusterMemberID int
+
+// must be called before any scraper
+func initClusterID() error {
+	clusterMemberNum := *clusterMemberNum
+	// special case for kubernetes deployment, where pod-name formatted at some-pod-name-1
+	// obtain memberNum from last segment
+	// https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2359
+	if idx := strings.LastIndexByte(clusterMemberNum, '-'); idx > 0 {
```
Comment (valyala) [inline lib/promscrape/config.go:67]:
```suggestion
	if idx := strings.LastIndexByte(s, '-'); idx >= 0 {
```

File: lib/promscrape/config.go [hunk 3]
```diff
@@ -49,12 +49,32 @@ var (
...
+		clusterMemberNum = clusterMemberNum[idx+1:]
+	}
+	i, err := strconv.ParseInt(clusterMemberNum, 10, 64)
+	if err != nil {
+		return fmt.Errorf("cannot parse clusterMemberNum=%q : %w", clusterMemberNum, err)
```
Comment (valyala) [inline lib/promscrape/config.go:72]:
```suggestion
		return fmt.Errorf("cannot parse -promscrape.cluster.memberNum=%q: %w", *clusterMemberNum, err)
```

File: lib/promscrape/config.go [hunk 4]
```diff
@@ -49,12 +49,32 @@ var (
...
+		"Can be specified as pod name of kubernetes statefulset - pod-name-1, where Num is numeric part of pod after -")
 	clusterReplicationFactor = flag.Int("promscrape.cluster.replicationFactor", 1, "The number of members in the cluster, which scrape the same targets. "+
 		"If the replication factor is greater than 2, then the deduplication must be enabled at remote storage side. See https://docs.victoriametrics.com/#deduplication")
 )
 
+var clusterMemberID int
+
+// must be called before any scraper
+func initClusterID() error {
+	clusterMemberNum := *clusterMemberNum
```
Comment (valyala) [inline lib/promscrape/config.go:63]:
```suggestion
	s := *clusterMemberNum
```

File: lib/promscrape/config.go [hunk 5]
```diff
@@ -49,12 +49,32 @@ var (
...
+	// special case for kubernetes deployment, where pod-name formatted at some-pod-name-1
+	// obtain memberNum from last segment
+	// https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2359
+	if idx := strings.LastIndexByte(clusterMemberNum, '-'); idx > 0 {
+		clusterMemberNum = clusterMemberNum[idx+1:]
+	}
+	i, err := strconv.ParseInt(clusterMemberNum, 10, 64)
```
Comment (valyala) [inline lib/promscrape/config.go:70]:
```suggestion
	n, err := strconv.ParseInt(s, 10, 64)
```

**Review by valyala — APPROVED** [review]

**Review by valyala — APPROVED** [review]

**Pull request merged**